### PR TITLE
Implement repositories, business logic and endpoints for roles and pe…

### PR DESCRIPTION
**Pull request by AntonPhk**
Implement repositories, business logic and endpoints for roles and permissions. Huge code refactoring (implement token service and inject it in all services), implement core repository for improving code style

### CODE REVIEW

File: src/repositories/core.py [hunk 1]
```diff
@@ -1,14 +1,85 @@
...
+    ):
+        stmt = select(self.model)
+
+        if sort_by:
+            sort_column = getattr(self.model, sort_by, None)
+            if sort_column:
+                if order_by == "asc":
+                    stmt = stmt.order_by(asc(sort_column))
+                elif order_by == "desc":
+                    stmt = stmt.order_by(desc(sort_column))
```
Comment (DzmitryTsekhanovitch) [inline src/repositories/core.py:41]: Line 37, redundant nesting. You can move `if sort_column` to same indent as `if sort_by`.

Comment (DzmitryTsekhanovitch) [reply src/repositories/core.py:41]: For order by you can use match/case

**Comment by DzmitryTsekhanovitch** [comment]
Great work, for me LGTM. left very minor comments.

File: src/api/routers/role.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+@router.get("/all", response_model=list[RoleWithPermissionsSchema])
+async def get_all_roles(
+    token: Annotated[str, Depends(oauth2_scheme)],
+    page: int = 1,
+    limit: int = 10,
+    order_by: str = "asc",
+    sort_by: Any = None,
+    role_service: RoleService = Depends(get_role_service),
+):
+    heh = await role_service.get_all_roles(
```
[the remark this answers is not in the document]
Comment (AntonPhk) [reply src/api/routers/role.py:36]: debug legacy :)

**Pull request merged**